core/src/main/java/io/substrait/dsl/SubstraitBuilder.java

```diff
@@ -57,35 +57,63 @@ public SubstraitBuilder(SimpleExtension.ExtensionCollection extensions) {
   }
 
   // Relations
+  public Aggregate.Measure measure(AggregateFunctionInvocation aggFn) {
+    return Aggregate.Measure.builder().function(aggFn).build();
+  }
+
+  public Aggregate.Measure measure(AggregateFunctionInvocation aggFn, Expression preMeasureFilter) {
+    return Aggregate.Measure.builder().function(aggFn).preMeasureFilter(preMeasureFilter).build();
+  }
+
   public Aggregate aggregate(
       Function<Rel, Aggregate.Grouping> groupingFn,
-      Function<Rel, List<AggregateFunctionInvocation>> measuresFn,
+      Function<Rel, List<AggregateFunctionInvocation>> aggsFn,
+      Rel input) {
+    Function<Rel, List<Aggregate.Measure>> measuresFn =
+        aggsFn.andThen(
+            functionInvocations ->
+                functionInvocations.stream().map(this::measure).collect(Collectors.toList()));
+    return aggregateM(groupingFn, measuresFn, input);
+  }
+
+  public Aggregate aggregateM(
```

Comment:
Thinking about this some more, I removed the the `aggregateM` methods and updated the `aggregate` methods to consume `Aggregate.Measure` instead of `AggregateFunctionInvocation`. I then updated the aggregate functions methods to return `Aggregate.Measure` as well.

Because the `aggregate` functions now consume `Aggregate.Measure`, we can now layer in additional utils to tweak the measure as needed like:
```java
            input ->
                List.of(
                    withPreMeasureFilter(
                        b.sum(input, 0), b.equal(b.fieldReference(input, 1), b.i32(42)))),
```